src/java/org/apache/cassandra/metrics/ChunkCacheMetrics.java

```diff
@@ -72,14 +73,14 @@ public void recordLoadFailure(long loadTime)
     }
 
     @Override
-    public void recordEviction()
+    public void recordEviction(int weight, RemovalCause cause)
     {
     }
 
     @Nonnull
     @Override
     public CacheStats snapshot()
     {
-        return new CacheStats(hits.getCount(), misses.getCount(), missLatency.getCount(), 0L, missLatency.getCount(), 0L, 0L);
+        return CacheStats.of(hits.getCount(), misses.getCount(), missLatency.getCount(), 0L, missLatency.getCount(), 0L, 0L);
```

Comment:
please validate this snapshot - I suppose that what is passed as `loadSuccessCount` is not what it should be...

Reply:
I think you are right. According to the javadoc:

```
[loadSuccessCount](https://www.javadoc.io/static/com.github.ben-manes.caffeine/caffeine/2.9.2/com/github/benmanes/caffeine/cache/stats/CacheStats.html#loadSuccessCount--)()
Returns the *number of times* [Cache](https://www.javadoc.io/static/com.github.ben-manes.caffeine/caffeine/2.9.2/com/github/benmanes/caffeine/cache/Cache.html) lookup methods have successfully loaded a new value.
```
So it should be a number of times, but we provide latency/time... I am not sure why that was done this way. 
It seems this was added in 4.0+ in [CASSANDRA-14628](https://issues.apache.org/jira/browse/CASSANDRA-14628)
@iamaleksey, any comments?

Reply:
My understanding is: missLatency is recorded whenever there is a cache miss and an entry is loaded into the cache. missLatency.getCount() returns number of times successful cache load happened. Which seems correct based on this understanding.

Reply:
you are right, though `missLatency.getCount()` is also used for load time, and in this case it is probably wrong; anyway, it is out of the scope

Reply:
I will open a follow-up ticket

Reply:
[CASSANDRA-18908](https://issues.apache.org/jira/browse/CASSANDRA-18908)